README.rst

```diff
@@ -0,0 +1,91 @@
+=========================
+OpenUpgrade Documentation
+=========================
+
+This branch aims to deal with the Openupgrade and openupgradelib documentation
+deployed on https://oca.github.io/OpenUpgrade/.
+
+- The main sources are present in the ``docsource`` folder.
+
+- The file ``build_openupgrade_docs`` does the following:
+
+    - fetch the migration coverage files present
+      in each branch of the OCA/Openupgrade project and copy them in the ``docsource``
+      folder.
+
+    - run the build via ``sphinx`` command and generate the output (html files)
+      in the ``docs`` folder.
+
+- The ``docs`` folder so contains autogenerated website that will be displayed
+  in the https://oca.github.io/OpenUpgrade/.
+
+Develop
+=======
+
+To contribute to the Openupgrade documentation:
+
+- Clone locally the current branch and create a dedicated branch
+
+.. code:: shell
+
+    git clone https://github.com/OCA/OpenUpgrade -b documentation
+    git checkout -b documentation-NEW-FEATURE
+
+- Install required components in a virtualenv:
+
+.. code:: shell
+
+    git clone https://github.com/odoo/odoo -b 16.0 --depth=1 ./src/odoo
+    virtualenv env --python=python3.10
+    ./env/bin/pip install -e ./src/odoo/
+    ./env/bin/pip install -r ./requirements.txt
+
+
+- Make changes in the docsource folder
+
+- Compile locally:
+
+.. code:: shell
+
+    . ./env/bin/activate
+    ./build_openupgrade_docs
+
+- Observe changes in a browser
+
+.. code:: shell
+
+    firefox ./docs/index.html
+
+- Cancel the changes in the ``docs`` folder, commit and push your work.
+
+.. code:: shell
+
+    git checkout ./docs/
```

Comment:
Could it be possible to skip this step?

I am thinking that `    ./build_openupgrade_docs` could build the docs by default in folder that is in the `.gitignore`, then github actions could call `    ./build_openupgrade_docs` with an options `--prod` that would build in the `docs` folder.

Then, I guess that this:
```
    firefox ./docs/index.html
```

Would become something like this:

```
For dev:
    firefox ./docs-dev/index.html
For prod:
    firefox ./docs/index.html
```

I did not test all of these, so I am sorry if this is not relevant.

Reply:
Why not ! Not sure it will be more simple. Feel free to make a PR. 
Note the scope of the current PR is to describe how it work. (this makes part of the documentation refactor).